### ISSUE TITLE
Regarding adding more features

### ISSUE THREAD

**Issue by gammaG2** [xofbd/wqu-weather-app#9]
I was wondering if this project would be continued for a time longer than our time at WQU session.

If this repo is planned to be around for other cohorts to join in as they complete their courses and contribute as a project, then it would be justified to add more complex features.

Proposed features if the above is in affirmative -

- Add a HTML or JS template to integrate new features.
- Air Quality Index (AQI)
- Satellite Image snapshot
- Weather Icons - Rainy, sunny, windy etc.
- Past observations and future prediction table for weather

**Comment by xofbd** [comment]
This is a good point. I have thought about it a bit and I am thinking of doing separate (but probably similar) projects for future cohorts. One idea, Is I can fork the project at an early point in git history. 

The problem with just keeping the same project is that the barrier of entry will be higher if it's a continuously evolving project. However, more complete projects can be done if we just continue with the first one.

In short, I am leaning towards new projects but I have not made a decision. I am open to ideas from everyone.

**Comment by gammaG2** [comment]
I thought about this from your perspective as well, before opening the issue. My take was that, should you have the time to manage 2 repos, then they could be split as follows -

1. **Repo A**- From scratch up until `html_templates` could be reset for each incoming batch.

1. **Repo B** - Taking the exit point of **Repo A** as entry point of **Repo B**, the continuing repo which would hopefully turn into some bigger project / OSS Tool. I suggested use of html_templates as even if features are added on a rolling basis, they can be cleanly integrated by template format.
Regarding the project, maybe this weather app deployed as a website feature on WQU-alumni.com < the domain is available at the time of writing >, maybe something else, if that is suitable. Or maybe a network resource to help in the coursework and the same could be deployed if it is made well enough over time.

**Comment by xofbd** [comment]
One thing to keep in mind is that this project came out organically. I didn't enter this session saying to myself that I wanted to do a group project. The whole idea came together in office hours. As such, I am in no obligation to maintain this project, especially _after_ the end of the session. For a little bit of detail, I work for Pragmatic Institute/TDI; we are under contract to create and deliver these WQU data science courses. Perhaps in the future (e.g., 2021) this project will become official.

With that said, I do like this plan. It may be that I step aside from **Repo B** and focus more on **Repo A**, especially once the cohort starts to ramp up.

An other opinions and thoughts from @gammaG2  or anyone else is welcomed!

**Comment by gammaG2** [comment]
I appreciate the fact that you did create this repo out of goodwill in the first place purely in the interest for dissemination of more knowledge beyond the existing charter of WQU/TDI.
It is also well understood that there is no obligation or incentive for you to continue this project after the unit is over and you shift your focus more towards the new intake.

The idea for **repo B** spinning out from unit 1 could be thought of as an entirely cohort managed repo, with your experienced inputs occasionally to guide the big picture direction, but for immediate bug fix or feature request it is in similar to a no warranty "as is" software since you would be focused more on the new cohort, or for that matter have any other engagement.

Having said all that it is entirely up to you to decide whether you want to engage in such an excercise afterall and your final call on this matter would be well accepted by all of us. We are grateful that you entertained queries beyond the scope of syllabus. There is no doubt you went out of your way to share knowledge about git, APIs and other libraries in python.

**Comment by xofbd** [comment]
We are sailing uncharted seas! I hope that lessons learned here will lead to a better planned project in the future. We'll see what happens in the coming weeks. I am leaning towards having a **repo A** and **repo B**. Stay tuned!
